tests/cpp/test_gpu3.cpp

```diff
@@ -8236,6 +8236,57 @@ TEST_F(NVFuserTest, FusionCpAsyncPredicateError) {
           ::testing::HasSubstr("unsupported use case of cp.async")));
 }
 
+TEST_F(NVFuserTest, DecoupledDomains) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // XX shape structure:
+  //
+  // domain 0: [I0, I1...    I4  I5} domain 1
+  //             \  /         \  /
+  //            merge         merge
+  //             /  \         /  \.
+  // domain 1: {I2  I3    ...I6  I7] domain 0
```

Comment:
I'm not sure how to interpret this comment. For example, what does `[I0, I1... I4 I5}` mean?

Reply:
> For example, what does `[I0, I1... I4 I5}` mean?

This means nothing. The two returned domains are:
```
[I0, I1, I6  I7]
```
and 
```
[I2,  I3, I4  I5]
```

Reply:
Isn't the second one `[I4, I5, I2, I3]`? Or I'm still missing something important. I still don't know how to interpret this comment.

Reply:
added a comment about what is domain 0 and what is domain 1

Reply:
> Isn't the second one `[I4, I5, I2, I3]`? Or I'm still missing something important. I still don't know how to interpret this comment.

I think it is just a matter of whether you read from top to bottom (`[I4, I5, I2, I3]`) or from left to right (`[I2, I3, I4, I5]`). Here I am taking the left to right approach.

Reply:
Hmm, not sure. I thought that the initial domain vector is `[I0, I1, I2, I3]`. And I thought that the two returned domains you referred to were generated by:

```
AbstractTensor dom0({id0, id1, id2, id3});
    AbstractTensor dom1 = dom0;
    dom0.merge(2);
    dom0.split(2, 256);
    dom1.merge(0);
    dom1.split(0, 256);
```

I'm assuming `I0` corresponds to `id0` (and `I1` to `id1`, etc). And also `dom0` in the code corresponds to `domain 0` in the comment. 

Both `domain 0` and `domain 1` start with `[I0, I1, I2, I3]`. `Domain 0` is then transformed with `merge(2)` and `split(2, 256)`, resulting in `[I0, I1, I2*I3/256, 256]`, where `I2*I3/256` and `256` corresponds to `I6` and `I7`, respectively.

Since `dom1` has `merge(0)` followed by `split(0, 256)`, I thought `dom1` should be transformed to `[I0*I1/256, 256, I2, I3]`, and I assumed `I0*I1/256` and `256` correspond to `I4` and `I5`.

Please let me know where I misunderstand.

Reply:
Oh, I see the confusion. Your understanding is correct. My naming for I2, I3 and I4, I5 was flipped. Updated the comment in the code.

Reply:
It's clear now. Thanks!